pkg/queryfrontend/request.go

```diff
@@ -133,6 +137,97 @@ func (r *ThanosQueryRangeRequest) String() string { return "" }
 // which is not used in thanos.
 func (r *ThanosQueryRangeRequest) ProtoMessage() {}
 
+type ThanosQueryInstantRequest struct {
```

Comment:
Do we have a way to use the same object for instant and range queries? That way we wouldn't have to duplicate code for these two use cases.

Reply:
I prefer to split the two types. Having them in the same struct is okay but we need a flag to maintain whether it is instant query and range query, and check the flag when needed.
Compared to having two types, this is not that clear and require other code, too.

Reply:
Got it, makes sense 👍